recipes-bsp/u-boot/u-boot_%.bbappend

```diff
@@ -44,22 +44,16 @@ OMNECT_BOOTLOADER_CHECKSUM_FILES_GLOB_IGNORE = "${OMNECT_THISDIR_SAVED}/u-boot/.
 # don't include files which are generated on build:
 OMNECT_BOOTLOADER_CHECKSUM_FILES_GLOB_IGNORE += "${OMNECT_THISDIR_SAVED}/u-boot/redundant-env-fragment.cfg"
 
-
-# copy configuration fragment from template, before SRC_URI is checked
-do_fetch:prepend() {
-    import os
-    cfg_file = d.getVar('OMNECT_THISDIR_SAVED', True) + d.getVar('PN', True) + "/" + "redundant-env-fragment.cfg"
-    cfg_file_template = cfg_file + ".template"
-    os.system("cp " + cfg_file_template + " " + cfg_file)
-}
-
```

Comment:
@JanZachmann @HarryWaschkeit I had to refactor that, because in scarthgap recipe parsing resulted in a race between `redundentand-env-fragment.cfg` exists or net yet. (In cicd it will never exist, in dobi it may (not clear why yet).)

i refactored it in do_configure:prepend below and https://github.com/omnect/meta-omnect/pull/422/files#diff-e486285ac515735595835791f1263c9aefa8d0913f03906ac1488932388cec56

question is: do we want this templated handling? it seems it was always a little bit overengineered from the beginning. we currently use it only for raspberry pi. 
e.g. offsets are computed here: https://github.com/omnect/meta-omnect/blob/02de8f089a8cc4a831bfef9a4567936c66332482/conf/distro/include/omnect-os-distro.conf#L92-L96

but tauri currently uses a static config. and the env lies before boot partition there.
maybe we can reuse the mechanism for tauri with scarthgap, but then we can not e.g. update from kirkstone->scarthgap via adu easily.

maybe it makes more sense to have a static configuration for each machine and leave the partition layout as it (kirkstone).

Reply:
For sure @HarryWaschkeit is the better addressee for that question but from what you write a static, per device, config is less complex and better maintainable + readable. So from my point of view I would go for that change.

Reply:
I need to understand first why environment regions need to be placed behind boot partition.
naively thinking I would expect that bootloader doesn't mind where boot partition starts, so I would expect it to be possible moving the boot partition a bit to make room for the two u-boot environments.
if that was possible, there would be no necessity at all to compute these offsets.

otoh, it looks completely ok to me to replace the templating with fixedly defined offsets, if the boot partition needs to be enlarged for whatever reason, it simply must be kept in mind to adapt also these offsets.

Reply:
> I need to understand first why environment regions need to be placed behind boot partition.

imho they don't. would have to test that though. if we change that kirkstone->scarthgap should be prohibited though.

Reply:
I would vote for the latter: do it right away in krikstone to have things in place for upgrade